src/Jaeger.Core/Propagation/TextMapCodec.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using OpenTracing.Propagation;
+
+namespace Jaeger.Core.Propagation
+{
+    public class TextMapCodec : Codec<ITextMap>
+    {
+        /// <summary>
+        /// Key used to store serialized span context representation.
+        /// </summary>
+        private const string SpanContextKey = "uber-trace-id";
+
+        /// <summary>
+        /// Key prefix used for baggage items.
+        /// </summary>
+        private const string BaggageKeyPrefix = "uberctx-";
+
+        private static readonly PrefixedKeys Keys = new PrefixedKeys();
+
+        private readonly string _contextKey;
+        private readonly string _baggagePrefix;
+        private readonly bool _urlEncoding;
+
+        public TextMapCodec(bool urlEncoding)
+            : this(new Builder().WithUrlEncoding(urlEncoding))
+        {
+        }
+
+        private TextMapCodec(Builder builder)
+        {
+            _urlEncoding = builder.UrlEncoding;
+            _contextKey = builder.SpanContextKey;
+            _baggagePrefix = builder.BaggagePrefix;
+        }
+
+        public override string ToString()
+        {
+            StringBuilder buffer = new StringBuilder();
```

Comment:
I ported it from Java. I have to use this special logic for `_urlEncoding` as a bool.ToString() returns `True` in C# vs `true` in Java. It would get less readable due to that. Would you still prefer the interpolation syntax?

Reply:
I also find it ugly that bools are upper-case in C# (especially since the idea is that it's XML compliant ant XML specifies it lower-case...). I assume `$"TextMapCodec{{contextKey={_contextKey},baggagePrefix={_baggagePrefix},urlEncoding={_urlEncoding.ToString().ToLowerInvariant()}}}"` to still be faster than to use a `StringBuilder` with a lot of Appends, but I haven't tested it.

Maybe we could extract the `var urlEncodedLower = _urlEncoding.ToString().ToLowerInvariant()` to keep the interpolated string readable. I at least find interpolation syntax easier readable than `StringBuilder` syntax.